// SPDX-License-Identifier: BSD-3-Clause-Clear
//
// Copyright (c) 2012-2013 ARM Limited
// All rights reserved
// Authors: Matteo Andreozzi
syntax = "proto2";
// Put all the generated messages in a namespace
package TrafficProfiles;

enum Command {
    INVALID   = 0;
    NONE      = 1;
    READ_REQ  = 2;
    WRITE_REQ = 3;
    READ_RESP = 4;
    WRITE_RESP= 5;
}

// packet data structure generated by the Traffic Profile Generator
message Packet {
  required Command cmd = 1;
  required uint64 time = 2;
  required uint64 addr = 3;
  required uint64 size = 4;
  required string master_id = 5;
  optional uint64 id = 6;
  required uint64 uid = 7 [default = 0];
  optional uint64 stream_id = 8;
  optional uint32 iommu_id = 9;
}
